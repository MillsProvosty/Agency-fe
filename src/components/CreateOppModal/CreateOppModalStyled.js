import styled from "styled-components";

export const CreateOppForm = styled.form`
  margin: 50px auto;
  height: auto;
  background-color: white;
  border: 1px solid #37474e;
  border-radius: 8px;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  width: 500px;
  padding: 20px;
  font-family: "Quicksand", sans-serif;
  color: #7a86cb;

  @media screen and (max-width: 375px) {
    margin: 130px 4px;
    height: auto;
    width: 98%;
    padding: 10px;
  }

  @media screen and (display-mode: standalone) {
    margin: 130px 4px;
    height: auto;
    width: 98%;
    padding: 10px;
  }
`;


export const PTag = styled.p`
  font-size: 2em;
`;

export const Input = styled.input`
  border-radius: 5px;
  font-size: 1.5em;
  height: 2em;
  border: 1px solid #37474e;
  padding: 5px;
  width: 300px;
  font-family: "Quicksand", sans-serif;
  margin-top: 10px;
`;

export const TextArea = styled.textarea`
  border-radius: 5px;
  font-size: 1.5em;
  height: 3em;
  border: 1px solid #37474e;
  padding: 5px;
  width: 300px;
  font-family: "Quicksand", sans-serif;
  margin-top: 10px;
`;

export const Button = styled.button`
  border-radius: 5px;
  font-size: 1.5em;
  height: 2em;
  padding: 5px;
  width: 300px;
  font-family: "Quicksand", sans-serif;
  margin-top: 10px;
  border: 2px solid white;
  color: white;
  background-color: #7a86cb;
  margin-top: 30px

  :hover {
    border: 2px solid #7a86cb;
    color: #7a86cb;
    background-color: white;
  }
`;
