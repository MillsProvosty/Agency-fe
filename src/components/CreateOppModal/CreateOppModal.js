import React, { useState, useEffect } from "react";
import { Link } from "react-router-dom";
import { validateCreateOpp } from "../../hooks/signInFormValidationRules";
import { useCreateOppForm } from "../../hooks/useForm";
import { setUserOpportunities, addUserOpp, setAllOpps } from "../../actions/";
import { postAnOpportunity } from "../../util/apiCalls";
import { connect } from "react-redux";
import { CreateOppForm, PTag, Input, TextArea, Button } from './CreateOppModalStyled'

export const CreateOppModal = props => {
  const [disabled, setDisabled] = useState(true);
  const [newValue, setNewValue] = useState(false)
  const { values, handleChange } = useCreateOppForm(validateCreateOpp);

  function setSetDisabled() {
    if (!values.error) {
      setDisabled(false);
    } else {
      setDisabled(true);
    }
  }

  const createOpp = async () => {
    let iterable;
    
    console.log(props.opportunities['0'])

    if (props.opportunities['0'] !== undefined && props.opportunities['0'].length > 1) {
      console.log('fuck', props.opportunities['0'].length)
      console.log('shit', props)
      iterable = props.opportunities[0];
      console.log("up", iterable);
    } else {
      iterable = props.opportunities;
      console.log("bitches", iterable);
    }

    let newOpp = await postAnOpportunity(props.user.id, values);
    let iterableSent = [...iterable, newOpp]
    console.log(iterableSent)
    await props.setAllOpps(iterableSent);
  };

  useEffect(() => {
    console.log("setNewValue")
    setNewValue(true)
  }, [props.opportunities]);

  useEffect(() => {
    validateCreateOpp(values);
    if (!values.error) {
      setSetDisabled();
    }
  }, [values]);

  return (
    <CreateOppForm>
      <PTag>Add an Opportunity For Our Community!</PTag>
      <Input
        type="text"
        placeholder="Enter a Title"
        name="title"
        value={values.title || ""}
        onChange={handleChange}
        autoComplete="off"
        required
      />
      <Input
        type="text"
        placeholder="Enter an Amount of Time"
        name="time"
        value={values.time || ""}
        onChange={handleChange}
        autoComplete="off"
        required
      />
      <Input
        type="text"
        placeholder="Enter an Address for the Event"
        name="address"
        value={values.address || ""}
        onChange={handleChange}
        autoComplete="off"
        required
      />
      <Input
        type="text"
        placeholder="Enter a Type: i.e. Physical labor"
        name="type"
        value={values.type || ""}
        onChange={handleChange}
        autoComplete="off"
        required
      />
      <TextArea
        type="text"
        placeholder="Enter a Description"
        name="description"
        value={values.description || ""}
        onChange={handleChange}
        autoComplete="off"
        required
      />
      <Link to="/schedule">
        <Button id='create-opp' disabled={disabled} onClick={createOpp}>
          Submit!
        </Button>
      </Link>
    </CreateOppForm>
  );
};

export const mapStateToProps = state => ({
  user: state.user,
  opportunities: state.opportunities
});

export const mapDispatchToProps = dispatch => ({
  addOpp: opp => dispatch(addUserOpp(opp)),
  setAllOpps: opportunities =>
  dispatch(setUserOpportunities(opportunities))
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(CreateOppModal);
