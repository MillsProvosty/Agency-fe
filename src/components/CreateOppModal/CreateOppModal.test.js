import React from 'react';
import ReactDOM from 'react-dom';
import CreateOppModal from './CreateOppModal';


it('renders without crashing', () => {
    expect(true).toEqual(true)
  });