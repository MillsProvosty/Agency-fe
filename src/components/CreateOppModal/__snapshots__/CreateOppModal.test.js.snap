// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateOppModal should match snapshot 1`] = `"wrapper"`;
